flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -439,6 +436,10 @@ public boolean triggerCheckpoint(long timestamp, boolean isPeriodic) {
 			triggerCheckpoint(timestamp, checkpointProperties, null, isPeriodic, false);
 			return true;
 		} catch (CheckpointException e) {
+			long latestGeneratedCheckpointId = getCheckpointIdCounter().get();
```

Comment:
Can you explain why we actually need to get the latest id again from the counter and cannot just remember it in a field when we do `checkpointID = checkpointIdCounter.getAndIncrement();` in `triggerCheckpoint(...)` as part of and `checkpointIdCounter.setCount(nextCheckpointId);` in `restore(...)`?

Reply:
I think to get the latest generated checkpoint id is the simplest and most direct way. In this way, we do not consider any potential problems:

- The concurrent problem, although at present, there is no risk to introduce a field to store, but based on Checkpoint IDCounter, there is no need to worry about this problem at all.
- Introducing a global field will bring potential misuse risks and hidden dangers for subsequent use in other logic in the future.